### PR TITLE
exporting: add curation check

### PR DESCRIPTION
**Pull request by nooraangelva**
#2397

### CODE REVIEW

File: backend/inspirehep/oai/utils.py [hunk 1]
```diff
@@ -37,11 +37,15 @@ def has_cern_affiliation(record):
...
     affiliations_values = get_value(record, "authors.affiliations.value", default=[])
     affiliations = chain.from_iterable(affiliations_values)
-    return any(
+    authors = get_value(record, "authors", default=[])
+    if any(
         affiliation
         for affiliation in affiliations
         if affiliation.lower().startswith("cern")
-    )
+    ) & any(author for author in authors if author["curated_relation"]):
```
Comment (MJedr) [inline backend/inspirehep/oai/utils.py:45]: this is causing an error in tests cause not all authors have curated_relation. Instead you have to do `author.get('curated_relation`),

File: backend/inspirehep/oai/utils.py [hunk 2]
```diff
@@ -37,11 +37,19 @@ def has_cern_affiliation(record):
...
+    authors = get_value(record, "authors", default=[])
+    if any(
         affiliation
         for affiliation in affiliations
         if affiliation.lower().startswith("cern")
-    )
+    ) & any(
+        author
+        for author in authors
+        if get_value(author, "curated_relation", default=True)
```
Comment (MJedr) [inline backend/inspirehep/oai/utils.py:48]: don't need to use `get_value` here, this is a wrapper which is useful when you have to get a value of the nested field. Also, default should be `false` which you don't need to specify cause `None` (returned by default) is `false`.

File: backend/inspirehep/oai/utils.py [hunk 3]
```diff
@@ -37,11 +37,19 @@ def has_cern_affiliation(record):
...
+    ):
+        return True
+    else:
+        return False
```
Comment (drjova) [inline backend/inspirehep/oai/utils.py:52]:
```suggestion
    ) & any(
        author
        for author in authors
        if get_value(author, "curated_relation", default=True)
    ):
        return True
    return False
```

**Comment by MJedr** [comment]
I am not sure if your solution is correct. In the task Micha wrote to check `curated` and not `curated_relation` in authors:
`"to also check that curated=True in the record."`

**Comment by nooraangelva** [comment]
> I am not sure if your solution is correct. In the task Micha wrote to check `curated` and not `curated_relation` in authors:
> `"to also check that curated=True in the record."`

You are right. I had searched for the ```curated``` field from a few records and I didn't find it so I presumed it was the ```curated_relation``` that he meant. I searched a few more records now and found the correct field. I will fix this solution. Thank you for noticing my mishap, when I didn't. :)

File: backend/tests/unit/oai/test_utils.py [hunk 4]
```diff
@@ -56,30 +56,76 @@ def test_has_cern_accelerator_experiment_without_field():
...
+        ],
     }
     record = faker.record("lit", data)
-    assert True == has_cern_affiliation(record)
+    assert has_cern_affiliation(record)
+
+
+def test_has_cern_affiliation_with_currated():
+    data_not_currated = {
+        "curated": False,
```
Comment (MJedr) [inline backend/tests/unit/oai/test_utils.py:100]: can you add a test without `curated` key too?

**Review by MJedr — APPROVED** [review]

**Pull request merged**